src/derivative_operators/vector_algebraic_operations.jl

```diff
@@ -0,0 +1,63 @@
+
+function dot_product(A::AbstractArray{Array{T,1},N},B::AbstractArray{Array{T,1},N}) where {T<:Real,N}
```

Comment:
why AbstractArray{Array}?

Also, does LinearAlgebra.dot not handle this case?

Reply:
- `AbstractArray{Array}` is the structure I was referring to at the start.
   Each Matrix entry would look like `[f₁ f₂ f₃ ... ]` representing various components of the vector.

- Yeah, just checked in LinearAlgebra. So dot, cross & norm from there will work too, when used as element-wise 
  operations. Should I create these as wrappers for feasibility of use? Also, LinearAlgebra didn't have inplace operations 
  for these, maybe introduce them here as new functions?